New and updated docs for HIS Designer for Logic Apps and 3270 Design Tool (#935)
MicrosoftDocs/biztalk-docs: his/core/application-integration-importhostdefs.md

   - The Designer will present the metadata artifact generated from the COBOL copybook.

   :::image type="content" source="media/la-newproject-import-hostdef6.png" alt-text="Metadata artifact view in HIS Designer for Logic Apps":::

   - The Designer will also generate a Host Definition for the Copybook. This Host Definition will not include the entire provided copybook but only the fields and datatypes needed for the artifact to interact with the Mainframe Program. In other words, while the sample provided above is an entire program, the HIS Designer will extract only the information needed depending on the selected [Programming Model](choosing-the-appropriate-programming-model1.md).

   :::image type="content" source="media/la-newproject-import-hostdef7.png" alt-text="Host Definition parsed by HIS Designer for Logic Apps view":::

1. Select **Save All** to generate the HIDX.

## Importing a COBOL Host Definition (IMS)

CICS and IMS host Mission Critical programs but have different requirements. The following steps will allow importing a COBOL copybook into the HIS Designer for Logic Apps. This COBOL program follows the IMS Connect Programming model.

> 1. Select on the **Component node** and then **Import** and **Host Definition**.

Reviewer: ```suggestion
1.  In the left pane, open the component node's shortcut menu, and select **Import** > **Host Definition**.

     In the following example, the component node is named **NetCInt1**.
```

   :::image type="content" source="media/la-newproject-import-hostdef1.png" alt-text="Importing IMS Host Definitions in Visual Studio":::

> 1. In the Import System z COBOL source file dialog, Select **Browse**

Reviewer: ```suggestion
1. In the **Import System z COBOL Source File** box, select **Browse**.
```

---

> title: "Importing COBOL Host Definitions"

Reviewer: ```suggestion
title: Importing COBOL Host Definitions
```

> description: Importing COBOL Host Definitions for CICS, IMS and Host Files

Reviewer: ```suggestion
description: Importing COBOL host definitions for CICS, IMS, and host files.
```

> author: haroldcampos

Reviewer: ```suggestion
```

> ms.author: "hcampos"

Reviewer: ```suggestion
```

> ms.prod: "host-integration-server"

Reviewer: ```suggestion
ms.prod: host-integration-server
```

ms.topic: how-to

> ms.date: "10/17/2023"

Reviewer: ```suggestion
ms.date: 10/17/2023
```

---

# Importing COBOL Host Definitions

> When the original Host Integration Server TI Designer feature was introduced, it was provided with a full set of capabilities to design metadata artifacts from scratch with the assistance of a Mainframe Programmer. The ability to import Host Definitions in the format of a COBOL or RPG copybook was introduced to support scenarios to automate design and to reduce involvement of Mainframe Programmers. Over the years, this became the principal option to create Metadata artifacts.

Reviewer: ```suggestion
The original Host Integration Server TI Designer feature was introduced with the full capabilities for designing metadata artifacts from scratch with the assistance from a mainframe programmer. Later, the capability to import host definitions in COBOL or RPG copybook format was added to support scenarios for automating design and reducing involvement from mainframe programmers. Over the years, this capability became the principal option to create metadata artifacts.
```

## Prerequisites

> The Main prerequisite is to obtain the Host Definitions (copybooks) that will be imported in the HIS Designer for Logic Apps. The HIS Designer for Logic Apps supports COBOL and RPG copybooks.

Reviewer: ```suggestion
- Obtain the host definitions (copybooks) that you want to import into HIS Designer for Logic Apps. This designer supports COBOL and RPG copybooks.
```

> It is also important to understand the Programming model for the technology to integrate against (CICS or IMS). Both platforms have different requirements and ways to pass and receive information. Please get yourself familiar with the appropriate [Programming Model](choosing-the-appropriate-programming-model1.md) before importing the Host Definitions.

Reviewer: Avoid ["please"](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/please), except in specific use cases:

```suggestion
 - Understanding how the programming model works for the technology that you want to integrate against, which is CICS or IMS. Both platforms have different requirements and ways to pass and receive information. Make sure that you learn about the appropriate [programming model](choosing-the-appropriate-programming-model1.md) before you import the host definitions.
```

Author: Understood

## Preparing a COBOL Copybook

> - COBOL copybooks should follow the basic COBOL coding rules. The HIS Designer for Logic Apps enforces many of those. The main ones are the following:

Reviewer: ```suggestion
- COBOL copybooks should follow the basic COBOL coding rules. The HIS Designer for Logic Apps enforces many of those rules. The following table lists the main rules:
```

   |Columns  |Type  |Observation  |
   |---------|---------|---------|

>    |1–6     |     Sequence number    |   The Mainframe programmer should not enter anything in these positions.      |

Reviewer: ```suggestion
   | 1–6 | Sequence number | Don't enter anything in these positions. |
```

>    |7     |      Indicator   |    This can be used to code a comment (*). A slash (/) is also accepted.     |

Reviewer: ```suggestion
   | 7 | Indicator | Use to code a comment (*). A slash (/) is also accepted. |
```

>    |8-11     |   A Margin (Area A)      |  77 level numbers and 01 level numbers.       |

Reviewer: Not a sentence, so no period at the end:

```suggestion
   | 8-11 | A Margin (Area A) | 77 level numbers and 01 level numbers |
```

>    |12-72     |   B Margin (Area B)      |  These are reserved for 02 levels and upper.       |

Reviewer: ```suggestion
   | 12-72 | B Margin (Area B) | Reserved for 02 levels and higher |
```

>    |73-80     |   Identification      |    No definitions are allowed here.     |

Reviewer: ```suggestion
   | 73-80 | Identification | No definitions are allowed here. |
```

> - Verify that the COBOL or RPG copybook meets the requirement of the selected [Programming Model](choosing-the-appropriate-programming-model1.md).

Reviewer: ```suggestion
- Verify that the COBOL or RPG copybook meets the requirement of the selected [programming model](choosing-the-appropriate-programming-model1.md).
```

> - Verify the dots at the end of every line. Although with the COBOL latest versions this is not required, be prepared.

Reviewer: Be prepared for what?

```suggestion
- Verify the dots at the end of every line. Make sure they appear, even though COBOL's latest versions don't require this formatting.
```

Reviewer: @haroldcampos: Could you please clarify the original wording or confirm this rewrite? Thanks!

> - If there are REDEFINES in your copybook, talk to the Mainframe programmer to verify the definition that you will use if there is no discriminant available.

Reviewer: ```suggestion
- If there are REDEFINES in your copybook, have the mainframe programmer confirm the definition that you want to use, if no discriminant is available.
```

> - Remove any character other than the ones stated in the previews chartablet. Make sure that you have the right count of characters.

Reviewer: What's a "chartablet"?

Author: It was the previous chart or table...it got messed up...will change to table

> - Remove any character other than the ones stated in the previews chartablet. Make sure that you have the right count of characters.

Reviewer: ```suggestion
- Remove any character other than the ones stated in the previous table. Make sure that you have the right count of characters.
```

> - Remove any character other than the ones stated in the previews chartablet. Make sure that you have the right count of characters.

Reviewer: ```suggestion
- Remove any character other than the ones stated in the previous table. Make sure that you have the right count of characters.
```

> - Remove any character other than the ones stated in the previews chartablet. Make sure that you have the right count of characters.

Reviewer: ```suggestion
- Remove any character other than the ones stated in the previous table. Make sure that you have the right count of characters.
```

## Importing a COBOL Host Definition (CICS)

> The following steps will allow importing a COBOL copybook into the HIS Designer for Logic Apps. This COBOL program follows the CICS ELM Link [Programming Model](choosing-the-appropriate-programming-model1.md).

Reviewer: ```suggestion
The following steps show how to import a COBOL copybook into the HIS Designer for Logic Apps. This COBOL program follows the CICS ELM Link [programming model](choosing-the-appropriate-programming-model1.md).
```

> 1. Select on the Component node and then Import and Host Definition.

Reviewer: ```suggestion
1. In the left pane, open the component node's shortcut menu, and select **Import** > **Host Definition**.

    In the following example, the component node is named **NetCInt1**.
```

Author: Thank you. Didn't know what to write

   :::image type="content" source="media/la-newproject-import-hostdef1.png" alt-text="Importing CICS Host Definitions in Visual Studio":::

> 1. In the Import System z COBOL source file dialog, Select **Browse**

Reviewer: ```suggestion
1. In the **Import System z COBOL Source File** box, select **Browse**.
```

   :::image type="content" source="media/la-newproject-import-hostdef2.png" alt-text="Selecting Host Definitions in Visual Studio (CICS)":::

> 1. Select the Copybook to be imported

Reviewer: ```suggestion
1. Find and select the copybook to import.
```

   :::image type="content" source="media/la-newproject-import-hostdef3.png" alt-text="Selecting the Host Definitions from their directory (CICS)":::

> 1. The following is the COBOL program to be imported:

Reviewer: Not a step, so no numbering:

```suggestion
   The following example shows the COBOL program to import:
```

              MOVE SPACES       TO ACCT-INFO        OF ACCT-ARRAY(1)

              MOVE 25.00        TO CHK-OD-CHG       OF ACCT-ARRAY(1)
              MOVE 2000.00      TO CHK-OD-LIMIT     OF ACCT-ARRAY(1)
              MOVE 'SAV1234567' TO CHK-OD-LINK-ACCT OF ACCT-ARRAY(1)
              MOVE '10/31/2005' TO CHK-LAST-STMT    OF ACCT-ARRAY(1)
              MOVE 1            TO CHK-DETAIL-ITEMS OF ACCT-ARRAY(1)
              MOVE 41852.16     TO CHK-BAL          OF ACCT-ARRAY(1)

              MOVE 'CHK7896112' TO ACCT-NUM         OF ACCT-ARRAY(2)
              MOVE 'C'          TO ACCT-TYPE        OF ACCT-ARRAY(2)
              MOVE SPACES       TO ACCT-INFO        OF ACCT-ARRAY(2)

              MOVE 25.00        TO CHK-OD-CHG       OF ACCT-ARRAY(2)
              MOVE 2000.00      TO CHK-OD-LIMIT     OF ACCT-ARRAY(2)
              MOVE 'SAV7891234' TO CHK-OD-LINK-ACCT OF ACCT-ARRAY(2)
              MOVE '10/31/2005' TO CHK-LAST-STMT    OF ACCT-ARRAY(2)
              MOVE 1            TO CHK-DETAIL-ITEMS OF ACCT-ARRAY(2)
              MOVE 41852.16     TO CHK-BAL          OF ACCT-ARRAY(2)
           END-IF.

           EXEC CICS RETURN END-EXEC.

   ```  

> 1. Preview the Copybook to be imported

Reviewer: ```suggestion
1. Review the copybook to import:
```

   :::image type="content" source="media/la-newproject-import-hostdef4.png" alt-text="Dialog with pre-loaded Host Definition in Visual Studio":::

> 1. The Item Options dialog will pre-polulate the name of the artifact along with a Link-to-Program name. Select **Next**.

Reviewer: ```suggestion
1. After the **Item Options** box opens and populates the artifact name and the **Link-to-Program name** value, select **Next**.
```

   :::image type="content" source="media/la-newproject-import-hostdef5.png" alt-text="Dialog to select options for Item such as Methods, Data Tables, Structures or Unions":::

>    - The Designer will present the metadata artifact generated from the COBOL copybook.

Reviewer: ```suggestion
   The designer presents the metadata artifact that's generated from the COBOL copybook.
```

   :::image type="content" source="media/la-newproject-import-hostdef6.png" alt-text="Metadata artifact view in HIS Designer for Logic Apps":::

>    - The Designer will also generate a Host Definition for the Copybook. This Host Definition will not include the entire provided copybook but only the fields and datatypes needed for the artifact to interact with the Mainframe Program. In other words, while the sample provided above is an entire program, the HIS Designer will extract only the information needed depending on the selected [Programming Model](choosing-the-appropriate-programming-model1.md).

Reviewer: ```suggestion
   The designer also generates a host definition for the copybook. This host definition doesn't include the entire provided copybook, but only the fields and datatypes needed for the artifact to interact with the mainframe program. Although the previously provided sample is an entire program, the HIS Designer extracts only the information required based on the selected [programming model](choosing-the-appropriate-programming-model1.md).
```

   :::image type="content" source="media/la-newproject-import-hostdef7.png" alt-text="Host Definition parsed by HIS Designer for Logic Apps view":::

> 1. Select **Save All** to generate the HIDX.

Reviewer: ```suggestion
1. To generate the HIDX, select **Save All**.
```

## Importing a COBOL Host Definition (IMS)

> CICS and IMS host Mission Critical programs but have different requirements. The following steps will allow importing a COBOL copybook into the HIS Designer for Logic Apps. This COBOL program follows the IMS Connect Programming model.

Reviewer: ```suggestion
Both CICS and IMS host mission-critical programs, but each have different requirements. The following steps show how to import a COBOL copybook into the HIS Designer for Logic Apps. This COBOL program follows the IMS Connect programming model.
```

## Preparing a COBOL Copybook

- COBOL copybooks should follow the basic COBOL coding rules. The HIS Designer for Logic Apps enforces many of those. The main ones are the following:

   |Columns  |Type  |Observation  |
   |---------|---------|---------|
   |1–6     |     Sequence number    |   The Mainframe programmer should not enter anything in these positions.      |
   |7     |      Indicator   |    This can be used to code a comment (*). A slash (/) is also accepted.     |
   | 8-11 | A Margin (Area A) | 77 level numbers and 01 level numbers |
   |12-72     |   B Margin (Area B)      |  These are reserved for 02 levels and upper.       |
   |73-80     |   Identification      |    No definitions are allowed here.     |

- Verify that the COBOL or RPG copybook meets the requirement of the selected [Programming Model](choosing-the-appropriate-programming-model1.md).
- Verify the dots at the end of every line. Although with the COBOL latest versions this is not required, be prepared.
- If there are REDEFINES in your copybook, talk to the Mainframe programmer to verify the definition that you will use if there is no discriminant available.

> - Remove any character other than the ones stated in the previews chartablet. Make sure that you have the right count of characters.

Reviewer: ```suggestion
- Remove any character other than the ones stated in the earlier table. Make sure that you have the correct number of characters.
```

## Importing a COBOL Host Definition (CICS)

The following steps show how to import a COBOL copybook into the HIS Designer for Logic Apps. This COBOL program follows the CICS ELM Link [programming model](choosing-the-appropriate-programming-model1.md).

1. In the left pane, open the component node's shortcut menu, and select **Import** > **Host Definition**.

    In the following example, the component node is named **NetCInt1**.

   :::image type="content" source="media/la-newproject-import-hostdef1.png" alt-text="Importing CICS Host Definitions in Visual Studio":::

1. In the **Import System z COBOL Source File** box, select **Browse**.

   :::image type="content" source="media/la-newproject-import-hostdef2.png" alt-text="Selecting Host Definitions in Visual Studio (CICS)":::

> 1. Find and select the copybook to import.

Reviewer: ```suggestion
1. Find and select the copybook to import, and then select **Open**.
```

   :::image type="content" source="media/la-newproject-import-hostdef7.png" alt-text="Host Definition parsed by HIS Designer for Logic Apps view":::

1. To generate the HIDX, select **Save All**.

## Importing a COBOL Host Definition (IMS)

Both CICS and IMS host mission-critical programs, but each have different requirements. The following steps show how to import a COBOL copybook into the HIS Designer for Logic Apps. This COBOL program follows the IMS Connect programming model.

1.  In the left pane, open the component node's shortcut menu, and select **Import** > **Host Definition**.

     In the following example, the component node is named **NetCInt1**.

   :::image type="content" source="media/la-newproject-import-hostdef1.png" alt-text="Importing IMS Host Definitions in Visual Studio":::

1. In the **Import System z COBOL Source File** box, select **Browse**.

   :::image type="content" source="media/la-newproject-import-hostdef2.png" alt-text="Visual Studio configuration for Logic Apps and HIS":::

> 1. Select the Copybook to be imported. Select **Open**.

Reviewer: ```suggestion
1. Find and select the copybook to import, and then select **Open**.
```

   :::image type="content" source="media/la-newproject-import-hostdef8.png" alt-text="Selecting Host Definitions in Visual Studio (IMS)":::

>    - The following is the COBOL program to be imported:

Reviewer: ```suggestion
   The following example shows the COBOL program to import:
```

               10  IOTP-TIME                        PIC S9(7) COMP-3.
               10  IOTP-MSG-NUMBER                  PIC S9(4) COMP.
               10  FILLER                           PIC X(2).
           05  IOTP-MOD-NAME                        PIC X(8).
           05  IOTP-USER-ID                         PIC X(8).
   
       PROCEDURE DIVISION.
           ENTRY 'DLITCBL' USING IOTP-PCB.
           CALL 'CBLTDLI' USING GU IOTP-PCB INPUT-AREA.
   
              IF IOTP-STATUS = END-OF-MSG
                 DISPLAY 'IOTP-STATUS = END-OF-MSG'
              END-IF.
              IF IOTP-STATUS = QUEUE-EMPTY
                 DISPLAY 'IOTP-STATUS = QUEUE-EMPTY'
              END-IF.
              IF IOTP-STATUS NOT = ' '
                 DISPLAY 'CALL FAILED IOTP-STATUS = ' IOTP-STATUS
              END-IF.
   
           MOVE 777.12               TO ACCBAL OF OUTPUT-AREA.
   
           MOVE LENGTH OF OUTPUT-AREA TO LLO.
           CALL 'CBLTDLI' USING ISRT IOTP-PCB OUTPUT-AREA.
              IF IOTP-STATUS NOT = ' '
                 DISPLAY 'SEND FAILED IOTP-STATUS = ' IOTP-STATUS
              END-IF.
   
           GOBACK.

   ```

> 1. Preview the Copybook to be imported

Reviewer: ```suggestion
1. Review the copybook to import. When you're ready, select **Next**.
```

   :::image type="content" source="media/la-newproject-import-hostdef9.png" alt-text="Dialog with pre-loaded Host Definition in Visual Studio (IMS)":::

1. The Item Options dialog will pre-polulate the name of the artifact along with a Transaction ID. Select **Next**.

   :::image type="content" source="media/la-newproject-import-hostdef10.png" alt-text="Dialog to select options for Item such as Methods, Data Tables, Structures or Unions (IMS)":::

> 1. The Input Area dialog will appear. You will will need to check the Input Area of the Program. Select **Next**.

Reviewer: ```suggestion
1. After the **Input Area** box opens, review the information in the **Input area** section. When you're ready, select **Next**.
```

   :::image type="content" source="media/la-newproject-import-hostdef11.png" alt-text="IMS Host transaction input area":::

> 1. The Output Area dialog will appear. You will will need to check the Output Area of the Program. Select **Next**.

Reviewer: ```suggestion
1. After the **Output Area** box opens, review information in the **Output area** section. When you're ready, select **Next**.
```

   :::image type="content" source="media/la-newproject-import-hostdef12.png" alt-text="IMS Host transaction output area":::

1. The Return Value dialog will appear. You will will need to check the Return Value field for the Program. Select **Next**.

   :::image type="content" source="media/la-newproject-import-hostdef13.png" alt-text="Return value dialog for host transaction":::

> 1. The Data Tables and Structures dialog will appear. You will will need to check the groups that will become Data Tables and Structures for the Program. This sample program does not require this setting. Select **Next**.

Reviewer: ```suggestion
1. After the **Data Tables, Structures and Unions** box appears, review the groups to use for the data tables and structures in your application. When you're ready, select **Next**.

   For this sample application, the following example doesn't require this setting.
```

   :::image type="content" source="media/la-newproject-import-hostdef14.png" alt-text="Data Tables and Structures dialog":::

> 1. The LL Fields Area dialog will appear. You will will need to check the LL fields that need to be excluded from the transaction. Select **Next**.

Reviewer: ```suggestion
1. After the **LL Fields Area** box opens, review the LL fields that must be excluded from the transaction. When you're ready, select **Next**.
```

   :::image type="content" source="media/la-newproject-import-hostdef15.png" alt-text="LL Fields Area dialog":::

> 1. The ZZ Fields Area dialog will appear. You will will need to check the ZZ fields that need to be excluded from the transaction. Select **Next**.

Reviewer: ```suggestion
1. After the **ZZ Fields Area** box opens, review the ZZ fields that must be excluded from the transaction. When you're ready, select **Next**.
```

   :::image type="content" source="media/la-newproject-import-hostdef16.png" alt-text="ZZ Fields Area dialog":::

> 1. The TRANCODE Fields Area dialog will appear. You will will need to check the TRANCODE fields that need to be excluded from  the Program. Select **Finish**.

Reviewer: ```suggestion
1. After the **TRANCODE Fields Area** box opens, review the TRANCODE fields that must be excluded from your application. When you're ready, select **Finish**.
```

   :::image type="content" source="media/la-newproject-import-hostdef17.png" alt-text="TRANCODE Fields Area dialog":::

>    - The Designer will present the metadata artifact generated from the COBOL copybook.

Reviewer: ```suggestion
   The designer shows the metadata artifact generated from the COBOL copybook:
```

   
   :::image type="content" source="media/la-newproject-import-hostdef18.png" alt-text="IMS generated metadata view":::
   

>    The Designer will also generate a Host Definition for the Copybook. This Host Definition will not include the entire provided copybook but only the fields and datatypes needed for the artifact to interact with the Mainframe Program. In other words, while the sample provided above is an entire program, the HIS Designer will extract only the information needed depending on the selected [Programming Model](choosing-the-appropriate-programming-model1.md).

Reviewer: ```suggestion
   The designer also generates a host definition for the copybook. This host definition doesn't include the entire provided copybook, but only the fields and datatypes needed for the artifact to interact with the mainframe program. Although the previously provided sample is an entire program, the HIS Designer extracts only the information required based on the selected [programming model](choosing-the-appropriate-programming-model1.md).
```

   
   :::image type="content" source="media/la-newproject-import-hostdef19.png" alt-text="Host Definition generated by HIS Designer for Logic Apps":::

> 1. Select **Save All** to generate the HIDX.

Reviewer: ```suggestion
1. To generate the HIDX, select **Save All**.
```

> ## Importing a COBOL Host File Definition

Reviewer: I assume this section is for host files? The other sections have (CICS) and (IMS) in their names.

```suggestion
## Importing a COBOL Host File Definition (Host Files)
```

Reviewer: @haroldcampos: Please confirm that this edit is correct? Thanks!

> There are multiple types of IBM Host Files. Host files can exist in Mainframes or Midranges and have their own types and characteristics. With an increasing demand on modernizing or migrating Mainframe or midranges applications that use Host File data such as VSAM files to Azure, it is becoming more common to access this files and integrate them into Modern solutions.

Reviewer: - Spell out acronyms first time.

```suggestion
IBM host files have multiple types and can exist in mainframes or midrange systems. Each have their own types and characteristics. The demand is increasing to modernize or migrate mainframe and midrange applications that use host file data is increasing. One example is migrating virtual storage access method (VSAM) files to Azure. With this demand, the use case is becoming more common to access and integrate these files into modern solutions.
```

Reviewer: @haroldcampos: Please confirm that this edit is OK? Thanks!

> The following steps will allow importing a COBOL copybook into the HIS Designer for Logic Apps. This COBOL copybook represents a simple VSAM file. The Import wizard will create structures and unions. Once the copybook was imported, you will be able to create Tables  and assign them the right schemas.

Reviewer: ```suggestion
The following steps show how to import a COBOL copybook into the HIS Designer for Logic Apps. This COBOL copybook represents a simple VSAM file. The Import wizard creates structures and unions. After you import the copybook, you can create and assign tables to correct schemas.
```

> 1. Select on the **Component** node and then **Import** and **Host Definition**.

Reviewer: ```suggestion
1. In the left pane, open the component node's shortcut menu, and select **Import** > **Host Definition**.
```

   :::image type="content" source="media/la-newproject-import-hostdefhf1.png" alt-text="Importing Host Files Definitions in Visual Studio":::

> 1. In the Import System z COBOL source file dialog, Select **Browse**

Reviewer: ```suggestion
1. In the **Import System z COBOL Source File** box, select **Browse**.
```

   :::image type="content" source="media/la-newproject-import-hostdefhf2.png" alt-text="Selecting Host Definitions in Visual Studio (Host Files)":::

1. Select the Copybook to be imported

   :::image type="content" source="media/la-newproject-import-hostdefhf3.png" alt-text="Selecting the Host Definitions from their directory (Host Files)":::

> 1. The following is the COBOL program to be imported:

Reviewer: ```suggestion
   The following example shows the COBOL program to import:
```

   ```
         ******************************************************************        
         *HIS TRANSACTION DESIGNER EXPORT, 9.0
         *DATA DECLARATION GENERATED ON 9/23/2013 5:21:51 PM
         ******************************************************************        
         *LIBRARY NAME.............CustomerDatabaseZOS.HIDX
         *DESCRIPTION..............NONE AVAILABLE
         ******************************************************************        
          01  CUSTOMER-RECORD.
              05 CUSTOMER-NAME               PIC X(30).
              05 CUSTOMER-SSN                PIC X(9).
              05 CUSTOMER-ADDRESS.
                 10 CUSTOMER-STREET          PIC X(20).
                 10 CUSTOMER-CITY            PIC X(10).
                 10 CUSTOMER-STATE           PIC X(4).
                 10 CUSTOMER-ZIP             PIC 9(5).
              05 CUSTOMER-PHONE              PIC X(13).
              05 CUSTOMER-ACCESS-PIN         PIC X(4).
   ```  
 

> 1. Preview the Copybook to be imported and review the options for REDEFINEs, use of Importer defaults and/or Generate structure on host definition indents. Select **Next**.

Reviewer: ```suggestion
1. Review the copybook to import. Verify the selections for the following options: **REDEFINE**, **Use Importer defaults**, and **Generate structure on indents** for the host definition. When you're ready, select **Next**. 
```

   :::image type="content" source="media/la-newproject-import-hostdefhf4.png" alt-text="Dialog with pre-loaded Host Definition in Visual Studio (Host Files)":::

>    - The Designer will present the metadata artifact generated from the COBOL copybook. Remember this artifact is incomplete. You will need to create a Table or Tables that will reflect the Host Files.

Reviewer: ```suggestion
   The designer shows the metadata artifact generated from the COBOL copybook. This artifact is incomplete, so you must create one or more tables that reflect the host files.
```

   :::image type="content" source="media/la-newproject-import-hostdefhf6.png" alt-text="Metadata artifact view in HIS Designer for Logic Apps (Host Files)":::

> 1. Select the Tables directory and right Select **Add table**.

Reviewer: ```suggestion
1. In the node tree, open the **Tables** shortcut menu, and select **Add Table**.
```

   :::image type="content" source="media/la-newproject-import-hostdefhf4.png" alt-text="Dialog with pre-loaded Host Definition in Visual Studio (Host Files)":::

   The designer shows the metadata artifact generated from the COBOL copybook. This artifact is incomplete, so you must create one or more tables that reflect the host files.

   :::image type="content" source="media/la-newproject-import-hostdefhf6.png" alt-text="Metadata artifact view in HIS Designer for Logic Apps (Host Files)":::

1. In the component node tree, open the **Tables** shortcut menu, and select **Add Table**.

   :::image type="content" source="media/la-newproject-import-hostdefhf7.png" alt-text="Addint a Table to the Host File Definition":::

1. In the **Alias** property, rename the table. In the **Host File Name** property, enter the mainframe or midrange system name for the host file. In the **Schema** property, assign the previously imported schema.

   :::image type="content" source="media/la-newproject-import-hostdefhf8.png" alt-text="Entering Alias, Host File Name and Schema of the Host File":::

> 1.  To generate the HIDX. file, select **Save All**.

Reviewer: ```suggestion
1. To generate the HIDX. file, select **Save All**.
```

>      The following example shows the final view of the host files metadata artifact:

Reviewer: ```suggestion
   The following example shows the final view of the host files metadata artifact:
```
